packages/core/src/queries/user.ts

```diff
@@ -86,12 +85,30 @@ export const hasUserWithIdentity = async (connectorId: string, userId: string) =
 
 export const insertUser = buildInsertInto<CreateUser, User>(pool, Users, { returning: true });
 
-export const findTotalNumberOfUsers = async () => getTotalRowCount(table);
+const buildUserSearchConditionSql = (search: string) => {
+  const searchFields = [fields.primaryEmail, fields.primaryPhone, fields.username, fields.name];
+  const conditions = searchFields.map((filedName) => sql`${filedName} like ${'%' + search + '%'}`);
 
-const findUserMany = buildFindMany<CreateUser, User>(pool, Users);
+  return sql`${sql.join(conditions, sql` or `)}`;
+};
 
-export const findAllUsers = async (limit: number, offset: number) =>
-  findUserMany({ limit, offset });
+export const findTotalNumberOfUsers = async (search: string | undefined) =>
+  pool.one<{ count: number }>(sql`
+    select count(*)
+    from ${table}
+    ${conditionalSql(search, (search) => sql`where ${buildUserSearchConditionSql(search)}`)}
+  `);
+
+export const findAllUsers = async (limit: number, offset: number, search: string | undefined) =>
+  pool.many<User>(
+    sql`
+      select ${sql.join(Object.values(fields), sql`,`)}
+      from ${table}
+      ${conditionalSql(search, (search) => sql`where ${buildUserSearchConditionSql(search)}`)}
+      limit ${limit}
+      offset ${offset}
+    `
```

Comment:
same here for findMany?

Reply:
[LOG-1599](https://linear.app/silverhand/issue/LOG-1599)